### PR TITLE
added the default config tags to the measure tags, updated tests and …

### PR DESCRIPTION
**Pull request by logicnow-john**
…readme

### CODE REVIEW

File: lib/appoptics.js [hunk 1]
```diff
@@ -16,7 +16,6 @@ var urlParse = require('url').parse;
 var https = require('https');
 var http = require('http');
 var fs = require('fs');
-var extend = require('extend');
```
Comment (logicnow-john) [inline lib/appoptics.js:19]: removed this as it wasn't being used

File: lib/appoptics.js [hunk 2]
```diff
@@ -290,6 +289,7 @@ var flushStats = function appopticsFlush(ts, metrics) {
     // Valid format for parsing tags out: global-prefix.name#tag1=value,tag2=value
     // NOTE: Name can include the source
     var vals = measureName.split('#');
+    Object.assign(measure.tags, tags);
```
Comment (logicnow-john) [inline lib/appoptics.js:292]: wanted to use spread operator here but travis appears to check node version 4/6 (not available) and latest stable.

File: README.md [hunk 3]
```diff
@@ -162,7 +162,7 @@ config in the appoptics configuration section of the StatsD config file:
 }
 ```
 
-Once your config has been updated, all metrics submitted to AppOptics will include your defined tags.
+Once your config has been updated, all metrics submitted to AppOptics will include your defined tags in the request payload and along with the actual measure tags.
```
Comment (bmacnaughton) [inline README.md:165]: Maybe an example of how this configuration is set would be useful. I couldn't determine how this worked until I read the tests. (I'm not familiar with this code at all, so maybe it's my shortcoming.)

Wording is a little awkward. Maybe replace "and along with" with "in addition to"?

File: test/librato_tests.js [hunk 4]
```diff
@@ -10,8 +10,9 @@ const http = require('http');
 const config = {
   debug: false,
   appoptics: {
-    token: '-',
-    api: 'http://127.0.0.1:' + serverPort,
+      token: '-',
```
Comment (bmacnaughton) [inline test/librato_tests.js:13]: There probably should be at least one test that uses a config without `{host: '127.0.0.1'}` just to verify that there is no problem if there are no global tags are set.  That would also test that the global `host` tag can be suppressed by setting `config.appoptics.host = false` (as line 534 in appoptics.js says) or specifically `null` (as the code implements it).

While testing that should have been there all along it's now necessary to make sure that nothing breaks if no global tags are set.

**Comment by bmacnaughton** [comment]
this looks good to me. thanks for taking the time to make this work. 

i made a few comments that you might want to consider but i don't think are critical.

**Review by bmacnaughton — APPROVED** [review]

**Comment by mbeale** [comment]
@logicnow-john I'm going to close this PR.  We created a separate branch to add a config option for this behavior.

**Pull request closed without merging**